sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/AnalysisHelper.scala

```diff
@@ -222,15 +222,23 @@ trait AnalysisHelper extends QueryPlan[LogicalPlan] { self: LogicalPlan =>
   }
 
   /**
-   * Recursively transforms the expressions of a tree, skipping nodes that have already
+   * Recursively top-down transforms the expressions of a tree, skipping nodes that have already
```

Comment:
The doc intentionally not mention the traversal order, as caller of this method should not care about it. If they care, they should call the up or down version.

Reply:
Removed for the default version.

This was a problem to me that I didn't know the implementation details by reading the comment. I only realized it when writing code which took me some time. But probably it was only personal experience.